core/src/main/java/org/apache/calcite/sql2rel/RelStructuredTypeFlattener.java

```diff
@@ -303,9 +303,11 @@ protected int getNewForOldInput(int oldOrdinal) {
    * corresponding field post-flattening, and also returns its type.
    *
    * @param oldOrdinal Pre-flattening ordinal
+   * @param existingOffset offset already calculated the target column inside the oldOrdinal column.
+   *                       For unnested column, it should be 0.
    * @return Post-flattening ordinal and type
    */
-  protected Ord<RelDataType> getNewFieldForOldInput(int oldOrdinal) {
+  protected Ord<RelDataType> getNewFieldForOldInput(int oldOrdinal, int existingOffset) {
```

Comment:
Why protected and not private? Is it meant to be subclassed?

Reply:
Hmmm, to your question YES and I did this because in the past it is protected and it's not a final class. So I could not eliminate the possibility that somebody subclasses it.

And thanks man you do remind me that I need to keep the original function signature because someone may subclass it. if there is a @override annotation then this change is not backward compatible.

Reply:
Oh, I re-read the code and actually I did not delete the original method. I will make it private since I don't see any points that people will override it. Good catch!